pilot/pkg/bootstrap/server.go

```diff
@@ -660,22 +704,15 @@ func (s *Server) initSecureGrpcServerDNS(port string, keepalive *istiokeepalive.
 	if err != nil {
 		return err
 	}
+	s.GRPCDNSListener = l
 
 	opts := s.grpcServerOptions(keepalive)
 	opts = append(opts, grpc.Creds(tlsCreds))
+
 	s.secureGRPCServerDNS = grpc.NewServer(opts...)
 	s.EnvoyXdsServer.Register(s.secureGRPCServerDNS)
 
 	s.addStartFunc(func(stop <-chan struct{}) error {
-		go func() {
-			if !s.waitForCacheSync(stop) {
-				return
-			}
-			if err := s.secureGRPCServerDNS.Serve(l); err != nil {
-				log.Errorf("error from GRPC server: %v", err)
-			}
-		}()
-
 		go func() {
 			<-stop
 			s.secureGRPCServerDNS.Stop()
```

Comment:
seems this should be paired with where we actually run the listener?

Reply:
Separate PR ? Will need a round of cleanup for stuff that is specific to 1.5. 

The entire 'stop' code is too complicated and spread across too many go routines...